sov-modules/sov-modules-macros/src/module_info.rs

```diff
@@ -199,6 +204,20 @@ fn make_prefix_func(
     }
 }
 
+fn make_fn_address(module_ident: &proc_macro2::Ident) -> proc_macro2::TokenStream {
+    quote::quote! {
+        fn address() -> sov_modules_api::Address {
+            use sov_modules_api::Hasher;
+            let module_path = module_path!();
+
+            let mut hasher = <<Self::Context as sov_modules_api::Spec>::Hasher>::new();
+            hasher.update(module_path.as_bytes());
+            hasher.update(stringify!(#module_ident).as_bytes());
```

Comment:
Is there any way that a malicious module can cause a collision, or is there  a natural domain separator at the end of module path?

Reply:
Fixed, now we calculate it as a hash of the module the prefix (which has a domain separator).